src/rest-server/src/models/v2/job-attempt.js

```diff
@@ -54,52 +51,40 @@ if (sequelize && launcherConfig.enabledJobHistory) {
 
   const list = async (frameworkName) => {
     let attemptData = [];
-    let uid;
+    const encodedFrameworkName = encodeName(frameworkName);
 
     // get latest framework from k8s API
-    let response;
+    let framework;
     try {
-      response = await k8sModel.getClient().get(
-        launcherConfig.frameworkPath(encodeName(frameworkName)),
-        {
-          headers: launcherConfig.requestHeaders,
-        }
+      framework = await databaseModel.Framework.findOne({
+        attributes: ['snapshot'],
+        where: {name: encodedFrameworkName}}
       );
     } catch (error) {
-      logger.error(`error when getting framework from k8s api: ${error.message}`);
-      if (error.response != null) {
-        response = error.response;
-      } else {
-        throw error;
-      }
+      logger.error(`error when getting framework from database: ${error.message}`);
+      throw error;
     }
 
-    if (response.status === 200) {
-      // get UID from k8s framework API
-      uid = response.data.metadata.uid;
+    if (framework) {
       attemptData.push({
-        ...(await convertToJobAttempt(response.data)),
+        ...(await convertToJobAttempt(JSON.parse(framework.snapshot))),
         isLatest: true,
       });
-    } else if (response.status === 404) {
-      logger.warn(`could not get framework ${uid} from k8s: ${JSON.stringify(response)}`);
-      return {status: 404, data: null};
     } else {
-      throw createError(response.status, 'UnknownError', response.data.message);
-    }
-
-    if (isNil(uid)) {
+      logger.warn(`could not get framework ${encodedFrameworkName} from database.`);
       return {status: 404, data: null};
     }
 
-    const sqlSentence = `SELECT snapshot as data FROM framework_history WHERE ` +
-      `frameworkName = '${encodeName(frameworkName)}' ` +
-      `ORDER BY uid ASC;`;
-    const pgResult = (await sequelize.query(sqlSentence))[0];
+    const historyFrameworks = await databaseModel.FrameworkHistory.findAll({
+        attributes: ['snapshot'],
+        where: {frameworkName: encodedFrameworkName},
+        order: [['attemptIndex', 'ASC']],
+      }
```

Comment:
Do we need to dedup here?
Such as use the latest snapshot for the same framework and attemptIndex

Reply:
Use hash(frameworkName, attemptIndex, historyType) as primary key to solve this problem.